cpp/benchmarks/fixture/benchmark_fixture.hpp

```diff
@@ -32,8 +32,9 @@ inline auto make_cuda() { return std::make_shared<rmm::mr::cuda_memory_resource>
 inline auto make_pool_instance()
 {
   static rmm::mr::cuda_memory_resource cuda_mr;
-  static rmm::mr::pool_memory_resource pool_mr{&cuda_mr};
-  return std::shared_ptr<rmm::mr::device_memory_resource>(&pool_mr);
+  static auto pool_mr =
+    std::make_shared<rmm::mr::pool_memory_resource<rmm::mr::cuda_memory_resource>>(&cuda_mr);
```

Comment:
what is the difference, was the returned type incorrect?

Reply:
In old code, address of the static object `pool_mr` goes into shared_pointer. (pointer of a local variable should not be held in shared_pointer). when that shared pointer is destroyed, it tries to delete the pointer but it's a static object, we are getting the error.

In this new code, I created an `pool_memory_resource` object (in heap) and made the shared_pointer as static. So, this shared pointer object will live until program termination. When that shared pointer object is deleted at program end, the `pool_memory_resource` object will be deleted.